lib/puma/acme/middleware.rb

```diff
@@ -3,25 +3,34 @@
 module Puma
   module Acme
     # ACME challenge response handler for HTTP-01 challenges.
-    class Middleware < Sinatra::Base
+    class Middleware
       def initialize(app, manager:)
         @app = app
         @manager = manager
+      end
 
-        super(app)
+      def call(env)
+        dup._call(env)
```

Comment:
why `dup`?

Reply:
Because rack middleware isn't threadsafe, and ivars leak across threads.